base/src/commands.rs

```diff
@@ -14,6 +14,14 @@ pub trait TpmCommand: Marshalable {
     type RespHandles: Marshalable;
 }
 
+/// ============================================================================
```

Comment:
We do have a coding style against doing block comments like this though: https://github.com/tpm-rs/docs/blob/main/coding-style.md#comments

I don't mind them, and this seems like a good place to use them - the alternative is that it's hard to see at a glance which section of the specification the commands fall within. But since we have a styleguide rule against it, so we should probably follow it first.

You have a few options here:

## 1. Just don't have these block comments.

Pretty easy, just delete them.

## 2. Move each section into a submodule, and comment the submodule.

So then this file would look like this:

```rs
mod start_up;
pub use start_up::*;
mod testing;
pub use testing::*;
// ... etc. ...
```

Then in your submodules (for example, here's how `testing.rs` would look):

```rs
//! 9 Start-up [TPM2.0 1.83]

/// 9.2 _TPM_Init
pub struct InitCmd {}

/// 9.3 TPM2_Startup
#[repr(C)]
#[derive(Clone, Copy, PartialEq, Marshalable)]
pub struct StartupCmd {
    pub startup_type: TpmSu,
}
impl TpmCommand for StartupCmd {
    const CMD_CODE: TpmCc = TpmCc::Startup;
    type Handles = ();
    type RespT = ();
    type RespHandles = ();
}

/// 9.4 TPM2_Shutdown
pub struct ShutdownCmd {}
```

## 3. Seek an exception

I would not personally feel comfortable granting an exception to the rule against block comments, because I did not pen that rule. So if we want to go this route we'd have to ask that author to review.

---

Do note that technically the TPM2 specification is allowed to change the layout and numbering of the sections. So I'm not sure how useful having a bunch of section numbers in the Rust documentation is. But I also see your point in providing some structure to the command declarations here. So I'm indifferent on which path we go down, just tell me which you'd like and I'll help you out.

Reply:
You know, I say all this, but I don't really like Option (2) - so I'd avoid that one actually. Option (1) or Option (3) are okay in my book. Option (2) is a little confusing because it's not easy to tell just from the name which submodule a command would be in.

Reply:
I agree - option (2) would split all structure definitions into separate files - and some section just have a single function whereas other sections has 20 functions. As mentioned above I am currently sticking to (1), and moving the TPM reference in brackets [...] to each request. I will be pushing a suggestion momentarily 👍 🙂

Reply:
If I am not missing anything, I think option (2) can be made sensible if we split commands into modules similar to how they are split into sections in the TPM specs.

and if there is 1-1 mapping between modules and chapters, it will make things relatively simple to upgrade each part of the implementation whenever there is spec update.

Reply:
I'm indifferent, I'll leave it to the author.